### PR TITLE
Fix mouse cursor unwanted automatic hover selection without moving it

### PR DESCRIPTION
**Pull request by vhanla**
This will fix issues #975 #1352 #1563 

Changes Explanation:

TL;DR
OnMouseEnter verifies if mouse's pos has changed
OnMouseMove updates selected item status
Bonus:
ListBox_PreviewMouseDown updates selected item status on mouse down (special scenarios).

Since MouseEnter triggers before MouseMove, I use the first to keep details about results item which is in mouse's cursor area, this will be useful later on ListBox Preview Mouse Down.
OnMouseMove will actually change the selected item status after a real mouse move has occurred.

However, it might occur that as soon as you relaunch (`ctrl`+`space`) Wox, and mouse's cursor area is pointing an item you want to click, since it is not moving and current selected item is the previous one but is not the one we are clicking, then ListBox_Preview_MouseDown will handle that scenario, using details saved on MouseEnter, i.e. selecting the item we want to click before actually Wox (MainWindow.xaml.cs) executes its mouse events handling.

### CODE REVIEW

**Comment by vhanla** [comment]
Oops, AppVeyor failed, but error messages are unrelated to my changes, I only modified two files:
`Wox/ResultListBox.xaml` and `Wox/ResultListBox.xaml.cs`

Appveyor throws error about `Wox/Infrastructure/Image/ThumbnailReader.cs`

**Comment by riscie** [comment]
Can we do something to get this merged?

**Comment by jhdxr** [comment]
thanks for your contribution.
can you open a new PR targeting `dev` branch?

**Comment by 2tolik** [comment]
Any updates on this? Do checks fail still?

**Comment by CartCaved** [comment]
@jhdxr I remade the changes from @vhanla against the dev branch in PR https://github.com/Wox-launcher/Wox/pull/2368
